rpc/src/rpc.rs

```diff
@@ -1001,30 +1001,34 @@ impl JsonRpcRequestProcessor {
             } else if commitment.is_confirmed() {
                 // Check if block is confirmed
                 let confirmed_bank = self.bank(Some(CommitmentConfig::confirmed()));
-                if confirmed_bank.status_cache_ancestors().contains(&slot)
-                    && slot
+                if confirmed_bank.status_cache_ancestors().contains(&slot) {
+                    if slot
                         <= self
                             .max_complete_transaction_status_slot
                             .load(Ordering::SeqCst)
```

Comment:
I don't really get this check. Isn't it possible that we froze a bank on a higher fork and the requested slot is still not frozen/available?

Reply:
This transaction status writing happens asynchronously by a worker thread, so a block can be frozen or rooted and the status is still not written. In that case, we cannot construct the return data for the RPC call.

Reply:
A more involved and better fix could be to go look up the status from the in-memory state about the block if it's not written to blockstore yet.

Reply:
Got it, well this change already seems like a good improvement, but seems like we need the more involved fix (later) to fully close out the linked issue

Reply:
Yea. I'll leave it open, just link for now.